managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -138,6 +138,8 @@ public void initiate() {
                     duplicateBuffer = payloadProcessorHandle.getProcessedPayload();
                 }
             }
+            this.dataLength = duplicateBuffer.readableBytes();
+            this.ml.currentLedgerSize += (dataLength - originalDataLen);
```

Comment:
Is it better to just reset the `dataLength` if the payload processor presents?

```java
if (payloadProcessorHandle != null) {
    duplicateBuffer = payloadProcessorHandle.getProcessedPayload();
    dataLength = duplicateBuffer.readableBytes();
}
```

Reply:
Already fixed